test/test-network/systemd-networkd-tests.py

```diff
@@ -141,7 +141,10 @@ def setUpModule():
     if asan_options or lsan_options or ubsan_options:
         drop_in += ['SystemCallFilter=']
     if use_valgrind or asan_options or lsan_options or ubsan_options:
-        drop_in += ['MemoryDenyWriteExecute=no']
+        drop_in += [
+            'MemoryDenyWriteExecute=no',
+            'Environment=SYSTEMD_MEMPOOL=0',
```

Comment:
I'm not convinced that this is a good idea. Is mempool incompatible with the sanitizers? We *do* want to test both with mempool and without, because the bugs that can be caught are partially different in both cases. Always disabling mempool under sanitizer would cause us to catch fewer bugs. We should only do this if we get false positives otherwise.

Reply:
This is based on the discussion in #12731. See https://github.com/systemd/systemd/issues/12731#issuecomment-500777609.

But, currently, the test does not check valgrind results and we need to check them manually. So, I will drop the change at least now.

Reply:
I don't think @evverx's comment is entirely true. Valgrind does report "global" leaks, it just doesn't print details by default. To avoid those reports, we have support for freeing of mempools when VALGRIND=1. So I think it's good that you dropped this change from this PR, we should discuss it separately.

Reply:
I didn't say it was entirely true :-) As usual, it depends especially on what is considered "global" memory leaks. What I meant was that in some cases when global structures refer to some memory Valgrind doesn't report that the memory is definitely lost. It just says that it's still reachable. The commit where mempools were freed at the end was supposed to "fix" exactly those "still reachable" memory leaks, as far as I know, but then another commit, where mempools were disabled by default via `disable_mempool_c` in a few libraries, was introduced and made it hard to reason about what is going on :-)

Anyway, I agree we shouldn't tweak the variable to just make Valgrind happy. Ideally, both `SYSTEMD_MEMPOOL=0` and `SYSTEMD_MEMPOOL=1` should be tested but it's kind of expensive so I think at this point it would probably make sense to keep testing whatever is set by default.